main/webapp/modules/core/scripts/index/default-importing-sources/sources.js

```diff
@@ -34,6 +34,12 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 function ThisComputerImportingSourceUI(controller) {
   this._controller = controller;
 }
+
+// Function to check if the URL getting entered is valid or not 
+function isUrlValid(url) {
+  // regex for a valid URL pattern
```

Comment:
Is there a source for this? If so, I think worth commenting where this came from so we can track back if any issues